app/Http/Controllers/RouteController.php

```diff
@@ -0,0 +1,143 @@
+
+        $routes = Route::latest()->get();
+        return response()->json([RouteResource::collection($routes), 'Route fetched.']);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(StoreRouteRequest $request)
+    {
+        // The incoming request is valid...     
+
+        /*
+        $validator = $request->validate([
+            'name' => 'required|string|max:255',
+            'description' => 'required',
+            'path_to_route_image' => 'required',
+            'expected_time' => 'required',
+            'path_to_map_image' => 'required',
+            'path_to_character_image' => 'required'
+        ]);
+
+        */
+        
+        $imageName = Str::random(32).".".$request->path_to_character_image->getClientOriginalExtension();
+
+        //save Image in Storag folder
+        Storage::disk('public')->put($imageName, file_get_contents($request->path_to_character_image));
+
+
+        $route = Route::create([
+            'name' => $request->name,
+            'description' => $request->description,
+            'path_to_route_image' => $request->path_to_route_image,
+            'expected_time' => $request->expected_time,
+            'path_to_map_image' => $request->path_to_map_image,
+            // we store only the image name in DB
+            'path_to_character_image' => $imageName,
+         ]);
+
+         //$route->waypoints()->attach(['1', '2']);
+         
+        //return new SimpleRoute($route);
+
+        //return Json Response
+        return response()->json([
+            'message' => 'Route successfully created.'
+        ], 200);
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($routeId)
+    {
+            //return response()->json($routeId);
+        
+            $route = Route::find($routeId);
+            if (is_null($route)) {
+                return response()->json('Data not found', 404); 
+            }
+    
+            return response()->json([new RouteResource($route)]);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(StoreRouteRequest $request, $id)
+    {
+        // The incoming request is valid...
+        
+        /*
+        $validator = $request->validate([
+            'name' => 'required|string|max:255',
+            'description' => 'required',
+            'path_to_route_image' => 'required',
+            'expected_time' => 'required',
+            'path_to_map_image' => 'required',
+            'path_to_character_image' => 'required'
+        ]);
+        */
+
+        $route = Route::find($routeId);
```

Comment:
This would raise a nullpointer exception if there exists no route with the given id.

```suggestion
        $route = Route::findOrFail($routeId);
```